wp_api/src/login/url_discovery.rs

```diff
@@ -9,6 +9,35 @@ use super::WpApiDetails;
 
 const API_ROOT_LINK_HEADER: &str = "https://api.w.org/";
 
+#[uniffi::export]
+fn new_api_discovery(site_url: String) -> NewUrlDiscoveryResult {
+    todo!()
+}
+
+#[derive(Debug, uniffi::Record)]
+pub struct NewUrlDiscoveryResult {
+    pub user_input_attempt: NewUrlDiscoveryAttemptResult,
```

Comment:
Curious if this can (or should) be a `Result<NewUrlDiscoveryAttemptResult, UrlDiscoveryAttemptError>` – that'd avoid the `is_successful` property (consuming code needs to branch either on a switch statement or on `if is_successful` so I don't think increases the costs)

Reply:
`uniffi` doesn't let us use `Result`s in fields. They can only be used as return types for functions. The likely reason being, the errors get mapped to exceptions in other languages - as they don't all have a `Result` type that's similar to Rust.